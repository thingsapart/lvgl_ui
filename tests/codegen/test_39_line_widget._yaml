# Drawing a simple zig-zag line.
- type: line
  lv_line_set_points: [[{x: 5, y: 5}, {x: 70, y: 70}, {x: 120, y: 10}, {x: 180, y: 60}, {x: 230, y: 10}], 5]
  lv_line_set_y_invert: true
