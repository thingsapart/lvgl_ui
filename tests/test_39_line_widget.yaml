# Drawing a simple zig-zag line.
- type: line
  points: [[{x: 5, y: 5}, {x: 70, y: 70}, {x: 120, y: 10}, {x: 180, y: 60}, {x: 230, y: 10}]]
  y_invert: true
