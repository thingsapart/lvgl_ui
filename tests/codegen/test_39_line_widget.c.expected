/* AUTO-GENERATED by the 'c_code' backend */

#include "lvgl.h"
#include "c_gen/lvgl_dispatch.h" // For obj_registry_add
#include "data_binding.h"

void create_ui(lv_obj_t* parent) {
    // --- Static Arrays for LVGL properties ---
    static const lv_point_precise_t s_static_array_0[] = { x(5), x(70), x(120), x(180), x(230) };

    // unnamed: line_0 (line)
    lv_obj_t* line_0 = lv_line_create(parent);

    lv_line_set_points(line_0, s_static_array_0, 5);
    lv_line_set_y_invert(line_0, true);

}
